kms/acceptance/snippets_test.rb

```diff
@@ -0,0 +1,621 @@
+    @key_ring_name = @client.key_ring_path @project_id, @location_id, @key_ring_id
+
+
+    @asymmetric_sign_ec_key_id = SecureRandom.uuid
+    @client.create_crypto_key(@key_ring_name, @asymmetric_sign_ec_key_id,
+                              purpose:          :ASYMMETRIC_SIGN,
+                              version_template: {
+                                algorithm: :EC_SIGN_P256_SHA256
+                              },
+                              labels:           { "foo" => "bar", "zip" => "zap" })
+
+    @asymmetric_sign_rsa_key_id = SecureRandom.uuid
+    @client.create_crypto_key(@key_ring_name, @asymmetric_sign_rsa_key_id,
+                              purpose:          :ASYMMETRIC_SIGN,
+                              version_template: {
+                                algorithm: :RSA_SIGN_PSS_2048_SHA256
+                              },
+                              labels:           { "foo" => "bar", "zip" => "zap" })
+
+
+    @symmetric_key_id = SecureRandom.uuid
+    @client.create_crypto_key(@key_ring_name, @symmetric_key_id,
+                              purpose:          :ENCRYPT_DECRYPT,
+                              version_template: {
+                                algorithm: :GOOGLE_SYMMETRIC_ENCRYPTION
+                              },
+                              labels:           { "foo" => "bar", "zip" => "zap" })
+
+    @hsm_key_id = SecureRandom.uuid
+    @client.create_crypto_key(@key_ring_name, @hsm_key_id,
+                              purpose:          :ENCRYPT_DECRYPT,
+                              version_template: {
+                                algorithm:        :GOOGLE_SYMMETRIC_ENCRYPTION,
+                                protection_level: "HSM"
+                              },
+                              labels:           { "foo" => "bar", "zip" => "zap" })
+
+    @asymmetric_decrypt_key_id = SecureRandom.uuid
+    @client.create_crypto_key(@key_ring_name, @asymmetric_decrypt_key_id,
+                              purpose:          :ASYMMETRIC_DECRYPT,
+                              version_template: {
+                                algorithm: :RSA_DECRYPT_OAEP_2048_SHA256
+                              },
+                              labels:           { "foo" => "bar", "zip" => "zap" })
+  end
+
+  after :all do
+    @client.list_crypto_keys(@key_ring_name).each do |key|
+      if key.rotation_period || key.next_rotation_time
+        updated_key = {
+          name:               key.name,
+          rotation_period:    nil,
+          next_rotation_time: nil
+        }
+        update_mask = { paths: ["rotation_period", "next_rotation_time"] }
+        @client.update_crypto_key updated_key, update_mask
+      end
+
+      filter = "state != DESTROYED AND state != DESTROY_SCHEDULED"
+      @client.list_crypto_key_versions(key.name, filter: filter).each do |version|
+        @client.destroy_crypto_key_version version.name
+      end
+    end
```

Comment:
As a note: This does not fully clean up the fixtures. (There does not appear to be an API for deleting a key ring for example.) As a result, we probably want to rethink this when we move it to google-cloud-ruby, for example having a persistent key ring and a persistent set of keys (and a rake task to create them). Probably not worth doing this while in ruby-docs-samples because we'd have to replicate it across N projects, but google-cloud-ruby always runs acceptance tests against a single project so we can build out persistent fixtures in that one project when we move the samples.